_posts/2016-12-05-haskell-package-checklist.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Haskell package checklist
+---
+
+- Host on GitHub
+  - Easy for other developers to contribute.
+  - Integrates well with other services.
+  - More popular than other hosts.
+- Build with Stack
+  - Easiest way to manage dependencies.
+  - Ensures that your package will remain buildable.
+  - Makes difficult workflows easy, like `--profile` and `--coverage`.
+- Use `hpack`
+  - Uses YAML instead of custom package format.
+  - Avoids unnecessary boilerplate.
+  - Integrated directly into Stack.
+- Name package with `kebab-case`
+  - Keep everything lowercase to avoid confusion.
+  - Use hyphens to separate words.
+  - Keep it short and memorable.
+- Use Semantic Versioning
+  - SemVer is used by many other languages.
+  - It matches how developers generally think about versions.
+  - Hackage recommends the PVP, which adds ambiguity.
+- Choose a license
+  - Packages without a license can’t really be used.
+  - BSD-3-Clause is the most popular, followed by MIT.
+  - Include the license file as well.
+- Write a README
+  - This is how most people will become familiar with your package.
+  - Describe the problem the package solves.
+  - Provide at least one concrete example.
+- Keep a change log
+  - Use Git tags to mark versions.
+  - Reading diffs is not an acceptable way for users to discover changes.
+  - GitHub releases are a great place to put human-readable changes.
+- Write a `synopsis`
+  - This shows up when searching and viewing your package.
+  - Make it short, imperative, and descriptive.
+  - Use the same thing in the `description` prefixed with the package name.
```

Comment:
This will upset people, but I agree with _something_ braindead easy like this. Of course, my approach (refer to the README.md) upsets people even more.